quarkchain/cluster/master.py

```diff
@@ -1553,13 +1554,13 @@ def main():
     network.start()
 
     done_callback = None
-    if jsonrpc_enabled:
+    if public_jsonrpc_enabled:
         public_json_rpc_server = JSONRPCServer.start_public_server(env, master)
+        done_callback = lambda: (public_json_rpc_server.shutdown(),)
+
+    if private_jsonrpc_enabled:
         private_json_rpc_server = JSONRPCServer.start_private_server(env, master)
-        done_callback = lambda: (
-            public_json_rpc_server.shutdown(),
-            private_json_rpc_server.shutdown(),
-        )
+        done_callback = lambda: (private_json_rpc_server.shutdown(),)
 
     master.do_loop(done_callback)
```

Comment:
if both enabled, only one of the callbacks will be run

Reply:
Changed.